core/src/main/java/tc/oc/pgm/filters/MatchStateFilter.java

```diff
@@ -0,0 +1,24 @@
+package tc.oc.pgm.filters;
+
+import java.util.function.Predicate;
+import tc.oc.pgm.api.filter.query.MatchQuery;
+import tc.oc.pgm.api.match.Match;
+
+public class MatchStateFilter extends TypedFilter<MatchQuery> {
+
+  private final Predicate<Match> matchPredicate;
+
+  public MatchStateFilter(Predicate<Match> matchPredicate) {
```

Comment:
I don't get the obsession for a predicate here. Just use a set

Reply:
See my other comment, this constructor can then be `MatchPhaseFilter(MatchPhase)`

Reply:
Yes! Predicate was an old thing before i actually used all MatchPhases (was using method refrences instead like Match::isRunning etc.)